wp-multi-network/includes/classes/class-wp-ms-networks-api.php

```diff
@@ -0,0 +1,859 @@
+			'items'             => array(
+				'type'          => 'string',
+			),
+		);
+
+		$query_params['exclude'] = array(
+			'description'        => __( 'Ensure result set excludes specific IDs.', 'wp-multi-network' ),
+			'type'               => 'array',
+			'items'              => array(
+				'type'           => 'integer',
+			),
+			'default'            => array(),
+		);
+
+		$query_params['include'] = array(
+			'description'        => __( 'Limit result set to specific IDs.', 'wp-multi-network' ),
+			'type'               => 'array',
+			'items'              => array(
+				'type'           => 'integer',
+			),
+			'default'            => array(),
+		);
+
+		$query_params['offset'] = array(
+			'description'        => __( 'Offset the result set by a specific number of items.', 'wp-multi-network' ),
+			'type'               => 'integer',
+		);
+
+		$query_params['order']      = array(
+			'description'           => __( 'Order sort attribute ascending or descending.', 'wp-multi-network' ),
+			'type'                  => 'string',
+			'default'               => 'desc',
+			'enum'                  => array(
+				'asc',
+				'desc',
+			),
+		);
+
+		$query_params['orderby']    = array(
+			'description'           => __( 'Sort collection by object attribute.', 'wp-multi-network' ),
+			'type'                  => 'string',
+			'default'               => 'id',
+			'enum'                  => array(
+				'id',
+				'domain',
+				'path',
+				'domain_length',
+				'path_length',
+				'network__in',
+			),
+		);
+
+		$query_params['path'] = array(
+			'description'       => __( 'Limit result set to networks of specific path.', 'wp-multi-network' ),
+			'type'              => 'array',
+			'items'             => array(
+				'type'          => 'string',
+			),
+		);
+
+		$query_params['path_exclude'] = array(
+			'description'       => __( 'Ensure result set excludes specific path.', 'wp-multi-network' ),
+			'type'              => 'array',
+			'items'             => array(
+				'type'          => 'string',
+			),
+		);
+
+		/**
+		 * Filter collection parameters for the networks controller.
+		 *
+		 * This filter registers the collection parameter, but does not map the
+		 * collection parameter to an internal WP_Network_Query parameter. Use the
+		 * `rest_network_query` filter to set WP_Network_Query parameters.
+		 *
+		 * @since 2.2.0
+		 *
+		 * @param array $query_params JSON Schema-formatted collection parameters.
+		 */
+		return apply_filters( 'rest_network_collection_params', $query_params );
+	}
+
+
+	/**
+	 * Checks if the network can be read.
+	 *
+	 * @since 2.2.0
+	 *
+	 * @param WP_Network      $network Network object.
+	 * @param WP_REST_Request $request Request data to check.
+	 * @return bool Whether the network can be read.
+	 */
+	protected function check_read_permission( $network, $request ) {
+
+		if ( 0 === get_current_user_id() ) {
+			return false;
+		}
+
+		return current_user_can( 'manage_network' );
```

Comment:
There is no capability yet for checking whether the user can read a specific network, so here we need to use:

```php
switch_to_network( $network->id );
$allowed = current_user_can( 'manage_network' );
restore_current_network();

return $allowed;
```